src/masternodeman.cpp

```diff
@@ -502,29 +506,61 @@ CMasternode* CMasternodeMan::GetNextMasternodeInQueueForPayment(int nBlockHeight
         //make sure it has as many confirmations as there are masternodes
         if (mn.GetMasternodeInputAge() < nMnCount) continue;
 
-        vecMasternodeLastPaid.push_back(make_pair(mn.SecondsSincePayment(), mn.vin));
+        if (nBlockHeight < TIER_BLOCK_HEIGHT) {
+            vecMasternodeLastPaid.push_back(make_pair(mn.SecondsSincePayment(), mn.vin));
+        }
+        else {
+            vecMasternodeTiers[mn.tier].push_back(make_pair(mn.SecondsSincePayment(), mn.vin));
+        }
+        nCount++;
     }
 
-    nCount = (int)vecMasternodeLastPaid.size();
-
     //when the network is in the process of upgrading, don't penalize nodes that recently restarted
     if (fFilterSigTime && nCount < nMnCount / 3) return GetNextMasternodeInQueueForPayment(nBlockHeight, false, nCount);
 
-    // Sort them high to low
-    sort(vecMasternodeLastPaid.rbegin(), vecMasternodeLastPaid.rend(), CompareLastPaid());
+    uint256 blockHash = 0;
+    if (!GetBlockHash(blockHash, nBlockHeight - 100)) {
```

Comment:
what will happen if nBlockHeight < 100 ?
Use named constant instead of 100

Reply:
Created named constant. If blockHeight is less than 100, than hash will be left as 0. Also, it is their legacy logic (about -100 in blockHeight), since masternodes get rewards only after the 100 block.